unix/x0vncserver/XDesktop.cxx

```diff
@@ -1038,3 +1055,23 @@ bool XDesktop::setCursor()
   return true;
 }
 #endif
+
+// X11 clipboard data is available, let the clients know
+void XDesktop::handleXClipboardAvailable(bool available) {
+  server->announceClipboard(available);
+}
+
+// A client wants clipboard data
+void XDesktop::handleClipboardRequest() {
+  server->sendClipboardData(selection.getClipboardData().c_str());
+}
+
+// When a client says it has clipboard data, request it 
+void XDesktop::handleClipboardAnnounce(bool available) {
+  if (available) server->requestClipboard();
```

Comment:
Could we wait until another application requests it? It would avoid transferring things needlessly.

Reply:
Sorry for not being clear, but in PR summary I was talking about how current code issues `XConvertSelection()` commands immediately on a clipboard change on server side, without first calling `announceClipboard(true)` and waiting for `handleClipboardRequest()` call from VNC client. But I already have a solution to do it the 2nd way.

As for the `server->requestClipboard();` here, my main motivation was to start with a simple working solution. Retrieving the clipboard  from VNC client _after_ another app requests it means that we need to queue that request (and potentially others) and fulfill it later when data is available. As you said, it would would avoid avoid needless transfers, and I am hoping to implement it. 

Although, requesting it immediately means that when user selects 'Paste', the data is available immediately without potential lag. It also avoids the problem mentioned above about owning multiple selections, as the data will always be cached.

Reply:
I tried to implement this lazy loading, but it really complicates things. In addition to the issues mentioned above:
- Eventually transferred data can be larger then MaxCutText, so selection ownership needs to be released in that case
- Can't rely on `TXWindow` to generate `SelectionNotify` events, because `selectionRequest()` will only add the request to queue.

So, I plan to keep this simple for now. Because in many cases, data will be immediately transferred anyway (needing to cache data if one selection ownership is lost, clients doing unsolicited transfer, clients not supporting extended clipboard).

Reply:
Okay. We can have that as a possible future improvement.